src/lib/helpers.test.ts

```diff
@@ -944,23 +944,46 @@ describe("replaceString", () => {
 });
 
 describe("checkOverlap", () => {
-  test("returns true if first record is empty", () => {
+  test("should return false since all binding annotations/labels do not exist on the object", () => {
+    expect(checkOverlap({ key1: "", key2: "" }, { key1: "something" })).toBe(false);
+  });
+  test("should return false since all binding annotations/labels values do not match on the object values", () => {
+    expect(checkOverlap({ key1: "key1", key2: "key2" }, { key1: "value1", key2: "key2" })).toBe(false);
+  });
+  test("should return true since all binding annotations/labels keys and values match the object keys and values", () => {
+    expect(checkOverlap({ key1: "key1", key2: "key2" }, { key1: "key1", key2: "key2" })).toBe(true);
+  });
+
+  test("should return true since all binding annotations/labels keys exist on the object", () => {
+    expect(checkOverlap({ key1: "", key2: "" }, { key1: "key1", key2: "key2" })).toBe(true);
+  });
+
+  test("(Mixed) should return true since key and key value match on object", () => {
+    expect(checkOverlap({ key1: "one", key2: "" }, { key1: "one", key2: "something" })).toBe(true);
+  });
+  test("(Mixed) should return false since key1 value is differnet on object", () => {
+    expect(checkOverlap({ key1: "one", key2: "" }, { key1: "different", key2: "" })).toBe(false);
+  });
+  test("should return true since binding annotation/label since object contains binding key", () => {
+    expect(checkOverlap({ key1: "" }, { key1: "value1" })).toBe(true);
+  });
```

Comment:
```suggestion
```
Still need it?  Seems like it's just a lesser form of the test [a few lines higher up](https://github.com/defenseunicorns/pepr/pull/651/files#diff-f1e16476bc27e2629bbaf4e6097a2fbed45c908424b6d585b11c6204c6664695R957):
```
  test("should return true since all binding annotations/labels keys exist on the object", () => {
    expect(checkOverlap({ key1: "", key2: "" }, { key1: "key1", key2: "key2" })).toBe(true);
  });
```